pkg/pushsync/pushsync.go

```diff
@@ -164,36 +164,47 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 		bytes := chunkAddress.Bytes()
 		if dcmp, _ := swarm.DistanceCmp(bytes, p.Address.Bytes(), ps.address.Bytes()); dcmp == 1 {
 			if ps.topologyDriver.IsWithinDepth(chunkAddress) {
-				ctxd, canceld := context.WithTimeout(context.Background(), timeToWaitForPushsyncToNeighbor)
-				defer canceld()
 
-				chunk, err = ps.validStamp(chunk, ch.Stamp)
-				if err != nil {
-					return fmt.Errorf("pushsync valid stamp: %w", err)
-				}
+				return func() error {
```

Comment:
this is a bit.....ugly :D maybe just increment the metric in the individual error handling

Reply:
:(